### ISSUE TITLE
Possible fatal error in library

### ISSUE THREAD

**Issue by supuflounder** [adafruit/Adafruit_IO_Arduino#56]
I found the following code in AdafruitIO_Data.cpp
```
int AdafruitIO_Data::toRed()
{
  if(! _value)
    return 0;

  char r[] = "0x";
  strncat(r, toChar() + 1, 2);

  return (int)strtol(r, NULL, 0);
}
```
This makes no sense.  The **"0x"** will be placed in the const area (flash memory) of the processor.  It is a fixed constant string.  It makes no sense to concatenate anything to this.  In the unlikely case that the **"0x"** was placed in RAM, there is no control of where it is or what follows it, and therefore random data will be clobbered.

Furthermore, strncat does not create a NUL-terminated string, so if there is random data following the **{'0', 'x', '\x00'}** means that we would get the result
`{'0', 'x', R, R, ....}`
where R,R represents the two hex digits being copied, and .... represents some random data which happens to follow the "0x" in memory.

Not sure what toChar() + 1 means.  It suggests that the value does not start until the second byte of the value.  No comments suggest what is going on here.  The use of **+3** and **+5** in **toGreen** and **toBlue** is likewise uncommented.  The offsets suggest that the value is stored as ?RRGGBB, where ? is some unknown character (perhaps "#").

I find it odd that "red" is reported as an int, because "red" is **0xRR0000** where **RR** is the hex value representing red, if the result is 24-bit color.  It is something else if 16-bit color is used, because the 16-bit format is
`RRRRRGGGGGGBBBBB`
Not sure why it is returning an int value which is just 8 bits.  I would have thought **byte** (**unsigned char**) would be more appropriate But the **strncat** is definitely wrong.  The same conversion error using a constant pointer is in **toGreen** and **toBlue**.

The correct code would be
```
char r[5];
strcpy(r, "0x");
strncpy(&r[2], toChar() + 1, 2);   // #RRGGBB
r[4] = '\x00';
```
Better still, have a subroutine
```
protected:
   int cvColor(int offset)
      {
       char d[5];
       strcpy(d, "0x");
       strncpy(&d[2], toChar() + offset, 2);
       d[4] = '\x00';
       return (int)strtol(d, NULL, 0);
      }
```
Then the following calls could be used:
```
return cvColor(1); // in toRed
return cvColor(3); // in toGreen
return cvColor(5); // in toBlue
```
      joe

**Comment by brentru** [comment]
@supuflounder  Yeah, I was skimming over this yesterday while I was implementing these helpers in io-python-client and thought it was strange. 

I'm like your subroutine approach, I'll work on an implementation later today, make a PR, and tag you for review if that's ok?

Thank you for your continued eagle-eye'd dissection of this library.

**Comment by brentru** [comment]
Patched in latest release, `2.7.14`: https://github.com/adafruit/Adafruit_IO_Arduino/releases/tag/2.7.14
